src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1200,6 +1200,94 @@ public async Task TestSqlMetadataForInvalidConfigEntities()
+                Permissions: new[] { GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                Relationships: null,
+                Mappings: null
+                );
+
+            // creating an entity with invalid source object and adding relationship with an entity with invalid source
+            Entity entityWithInvalidSourceAndRelationship = new(
+                Source: new(null, EntitySourceType.Table, null, null),
+                Rest: null,
+                GraphQL: new(Singular: "publisher", Plural: "publishers"),
+                Permissions: new[] { GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                Relationships: new Dictionary<string, EntityRelationship>() { {"books", new (
+                    Cardinality: Cardinality.Many,
+                    TargetEntity: "Book",
+                    SourceFields: null,
+                    TargetFields: null,
+                    LinkingObject: null,
+                    LinkingSourceFields: null,
+                    LinkingTargetFields: null
+                    )}},
+                Mappings: null
+                );
+
+            configuration = configuration with
+            {
+                Entities = new RuntimeEntities(new Dictionary<string, Entity>()
+                    {
+                        { "Book", entityWithInvalidSource },
+                        { "Publisher", entityWithInvalidSourceAndRelationship}
+                    })
+            };
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(CUSTOM_CONFIG, configuration.ToJson());
+
+            FileSystemRuntimeConfigLoader configLoader = TestHelper.GetRuntimeConfigLoader();
+            configLoader.UpdateConfigFilePath(CUSTOM_CONFIG);
+            RuntimeConfigProvider configProvider = TestHelper.GetRuntimeConfigProvider(configLoader);
+
+            Mock<ILogger<RuntimeConfigValidator>> configValidatorLogger = new();
+            RuntimeConfigValidator configValidator =
+                new(
+                    configProvider,
+                    new MockFileSystem(),
+                    configValidatorLogger.Object,
+                    isValidateOnly: true);
+
+            ILoggerFactory mockLoggerFactory = TestHelper.ProvisionLoggerFactory();
+
+            try
+            {
+                await configValidator.ValidateEntitiesMetadata(configProvider.GetConfig(), mockLoggerFactory);
+            }
+            catch
+            {
+                Assert.Fail("The validation should not throw an exception.");
+            }
+
+            Assert.IsTrue(configValidator.ConfigValidationExceptions.Any());
+            List<string> exceptionMessagesList = configValidator.ConfigValidationExceptions.Select(x => x.Message).ToList();
+            Assert.IsTrue(exceptionMessagesList.Contains("The entity Book does not have a valid source object."));
```

Comment:
Should we confirm that we only return these messages once? What if the validation is adding erroneous messaging, should we confirm that this is the limit of the messaging?

Reply:
some messages can occur more than once if certain methods are called from multiple places or if different methods throw same error. for example `GetDatabaseObjectName` and `GetSchemaName` in SqlMetadatProvider are throwing the same error in case they dont find the mapping. 

This is because we try to capture all the error at once, this can be part of seperate PR where we refactor our code with better error messages.

Reply:
For this test, i asserted on all the unique exceptions thrown. 2 errors repeated because they're called from the above two methods i mentioned.

Reply:
When i run the cli using the config file created in this test, i see duplicates of the errors Aaron mentions. While I understand that the errors may be sourced from different places, this makes the errors pretty noisy if there are duplicates.. 
- "Table Definition for Book has not been inferred" -> appears twice
![image](https://github.com/Azure/data-api-builder/assets/6414189/cb8f01f3-de0b-435f-ab4a-1d7d8b631822)

Reply:
> This is because we try to capture all the error at once, this can be part of seperate PR where we refactor our code with better error messages.

understood that we can always add better messages.. but duplicate messages seem like a problem that can be addressed sooner?

Reply:
I suppose this behavior is consistent with how `dab validate` has operated thus far. open an issue to track enhancements such as this if not opened already.

Reply:
created #2131 . Don't think it would be a big change but yes out of scope of this PR.